src/sentry/monitors/clock_dispatch.py

```diff
@@ -22,6 +22,12 @@
 # This key is used to store the hashmap of Mapping[PartitionKey, Timestamp]
 MONITOR_TASKS_PARTITION_CLOCKS = "sentry.monitors.partition_clocks"
 
+# This key is used to record historical date about the volume of check-ins.
+MONITOR_VOLUME_HISTORY = "sentry.monitors.volume_history:{}"
+
+# We record 30 days worth of historical data for each minute of check-ins.
+MONITOR_VOLUME_RETENTION = timedelta(days=30)
```

Comment:
Since we keep keys every minute, this will be 43k keys. Could be a good reason to have our own cluster.